bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/BulkMutation.java

```diff
@@ -338,10 +328,10 @@ public synchronized void run() {
       ListenableFuture<List<MutateRowsResponse>> future = null;
       try {
         if (retryId == null) {
-          retryId = Long.valueOf(this.asyncExecutor.getRpcThrottler().registerRetry());
+          retryId = Long.valueOf(asyncExecutor.getRpcThrottler().registerRetry());
         }
-        MutateRowsRequest request = currentRequestManager.build();
-        future = asyncExecutor.mutateRowsAsync(request);
+        activeBatches.put(currentBatch.id, currentBatch);
```

Comment:
I don't think it's strictly necessary given the current code but I think it would be best to implement hashCode/equals for Batch since it's being used as a map key. Otherwise LGTM.